### DISCUSSION TITLE
Setting msg.value

### DISCUSSION THREAD

**Discussion by ssadler** [foundry-rs/foundry#3984] [Help]
Hi, I have some internal functions that read `msg.value` and expect for there to be a value. I looked but didn't find a way to set `msg.value`, is there one (apart from creating a public interface in between)? If not, is it considered an antipattern to have internal functions that read msg.value? If not, any plans to support setting `msg.value`? :slightly_smiling_face:

**Comment by foxnike** [comment]
you can try use `contractName.funName{value: 1 ether}()`  and `vm.prank()`  to test a payable function
